mita/openstack.py

```diff
@@ -147,15 +147,27 @@ def _get_image(self) -> NodeImage:
             logger.error(be)
             raise OpenStackDriverError("Encountered an unknown exception.")
 
-    def _get_flavor(self) -> NodeSize:
-        """Return the flavor reference."""
-        try:
-            return [f for f in self.driver_v2.list_sizes() if f.name == self.vm_size][0]
-        except IndexError:
-            raise ResourceNotFound("Flavor {} not found".format(self.vm_size))
-        except BaseException as be:  # noqa
-            logger.error(be)
-            raise OpenStackDriverError("Encountered an unknown exception.")
+    def _get_flavor(self, name: Optional[str] = None) -> NodeSize:
+        """
+        Return the NodeSize instance retrieved based on the provided name or vm_size.
+
+        Args:
+            name:   The name of the flavor to be retrieved
+
+        Return:
+            Reference to NodeSize
+
+        Raises:
+            ResourceNotFound exception when none resources found.
+        """
+        name = self.vm_size if name is None else name
+        logger.info("Gathering flavor details of %s", name)
```

Comment:
This logging is excessive, and the cephci logs are already massive at INFO level. Can we remove this `logger.info()` statement?